clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -181,11 +183,13 @@ private void process(final ListOffsetsApplicationEvent event) {
      * it is already a member.
      */
     private void processSubscriptionChangeEvent() {
-        if (!requestManagers.membershipManager.isPresent()) {
-            throw new RuntimeException("Group membership manager not present when processing a " +
-                    "subscribe event");
+        if (!requestManagers.heartbeatRequestManager.isPresent()) {
+            KafkaException error = new KafkaException("Group membership manager not present when processing a subscribe event");
```

Comment:
I expect this would only be the case of a bug in our code, so a `RuntimeException` that would fail the background thread (and therefore the consumer) seemed simple and enough. Is there a reason why we need all the logic to pass this failure to the app thread?